sympy/physics/continuum_mechanics/beam.py

```diff
@@ -482,39 +481,29 @@ def deflection(self):
         >>> b.apply_load(R2, 30, -1)
         >>> b.apply_load(120, 30, -2)
         >>> b.bc_deflection = [(10, 0), (30, 0)]
-        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> b.solve()
         >>> b.deflection()
         (4000*x/3 - 4*SingularityFunction(x, 0, 3)/3 + SingularityFunction(x, 10, 3)
             + 60*SingularityFunction(x, 30, 2) + SingularityFunction(x, 30, 3)/3 - 12000)/(E*I)
         """
+        
         x = self.variable
         E = self.elastic_modulus
         I = self.second_moment
-        if not self._boundary_conditions['deflection'] and not self._boundary_conditions['slope']:
-            return S(1)/(E*I)*integrate(integrate(self.bending_moment(), x), x)
-        elif not self._boundary_conditions['deflection']:
-            return integrate(self.slope(), x)
-        elif not self._boundary_conditions['slope'] and self._boundary_conditions['deflection']:
-            C3 = Symbol('C3')
-            C4 = Symbol('C4')
-            slope_curve = integrate(self.bending_moment(), x) + C3
-            deflection_curve = integrate(slope_curve, x) + C4
-            bc_eqs = []
-            for position, value in self._boundary_conditions['deflection']:
-                eqs = deflection_curve.subs(x, position) - value
-                bc_eqs.append(eqs)
-            constants = list(linsolve(bc_eqs, (C3, C4)))
-            deflection_curve = deflection_curve.subs({C3: constants[0][0], C4: constants[0][1]})
-            return S(1)/(E*I)*deflection_curve
-
-        C4 = Symbol('C4')
-        deflection_curve = integrate((E*I)*self.slope(), x) + C4
-
-        bc_eqs = []
-        for position, value in self._boundary_conditions['deflection']:
-            eqs = deflection_curve.subs(x, position) - value
-            bc_eqs.append(eqs)
-
-        constants = list(linsolve(bc_eqs, C4))
-        deflection_curve = deflection_curve.subs({C4: constants[0][0]})
+
+        slope_curve = integrate(self.bending_moment(), x) + self.C3
+        deflection_curve = integrate(slope_curve, x) + self.C4
```

Comment:
`C3` and `C4` can be evaluated by using `self._boundary_conditions['slope']` and `self._boundary_conditions['deflection']` respectively.